src/util/log.cc

```diff
@@ -553,6 +553,23 @@ std::ostream& operator<<(std::ostream& out, const std::exception_ptr& eptr) {
             } catch (...) {
                 out << ": " << std::current_exception();
             }
+        } catch (const std::nested_exception& ne) {
+           // if we got here, it means that the nested_exception is not mixed in
+           // with an object of type std::exception. It can happen in one of two
+           // cases:
+           // 1. The std::nested_exception was built and thrown using a normal
+           //    throw and not in the context of another catch with std::throw_with_nested
+           // 2. Somewhere in the code there is a throw of a type which doesn't derive from
+           //    std::exception and then it is rethrown using std::throw_with_nested.
+           out << ": " << "unknown nested type";
```

Comment:
Why is this an "unknown nested type"?  When we print ne.nested_ptr() below, doesn't it know it's own type? We have std::ostream& operator<<(std::ostream& out, const std::exception_ptr& eptr) which does exactly that.

Reply:
The resulting `std::nested_exception` type when generated properly with `std::throw_with_nested(T object)` is not a pure `std::nested_exception` but a **mix of two types** - a nested_type that *contains* the current exception, and the type
contained in the object. something like `class x : public T, public nested_exception {...` 
We can't account for every T in our code so we also account for objects derived from `std::exception` and then we print 
the what.
The "unknown nested type" print, indicates that we don't know the **whole** nested type, or more specifically, what T is and hence we can't print it.
About the test - fair enough, I will add a test instead the code in the commit message.

Reply:
Hmm, I think I see now what you meant about the "unknown nested type". Ok.